source/fuzz/transformation_swap_two_functions.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021 Shiyu Liu
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_swap_two_functions.h"
+
+#include "source/opt/function.h"
+#include "source/opt/module.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationSwapTwoFunctions::TransformationSwapTwoFunctions(
+    protobufs::TransformationSwapTwoFunctions message)
+    : message_(std::move(message)) {}
+
+TransformationSwapTwoFunctions::TransformationSwapTwoFunctions(uint32_t id1,
+                                                               uint32_t id2) {
+  message_.set_function_id1(id1);
+  message_.set_function_id2(id2);
+}
+
+bool TransformationSwapTwoFunctions::IsApplicable(
+    opt::IRContext* ir_context, const TransformationContext& /*unused*/) const {
+  auto func1_ptr = ir_context->GetFunction(message_.function_id1());
+  auto func2_ptr = ir_context->GetFunction(message_.function_id2());
+  if (!func1_ptr || !func2_ptr ||
+      func1_ptr->result_id() == func2_ptr->result_id())
+    return false;
+  return true;
+}
+
+void TransformationSwapTwoFunctions::Apply(
+    opt::IRContext* ir_context, TransformationContext* /*unused*/) const {
+  // Found the two functions in ir_context and swap their position.
+  // Offsets mark the relevant distance of the function from module().begin().
+  bool func1_found = false;
+  bool func2_found = false;
+
+  // Initialize the position (underlying: UptrVectorIterator<Function>)
+  opt::Module::iterator func1_it = ir_context->module()->begin();
+  opt::Module::iterator func2_it = ir_context->module()->begin();
+  for (auto& func : *ir_context->module()) {
+    if (func.result_id() == message_.function_id1()) func1_found = true;
+    if (func.result_id() == message_.function_id2()) func2_found = true;
+    // Once we found the target function, we stop increment iterator and thus
+    // after one iteration, func1_it and func2_it should be the iterator with
+    // their updated position.
+    // If we have not found (ie. found = false), we kept incrementing.
+    if (!func1_found) ++func1_it;
+    if (!func2_found) ++func2_it;
+  }
```

Comment:
I suggest using `std::find_if` or add another utility function to fuzzer_util.cpp/h that is like `FindFunction`, but returns an iterator (pointing to end if not found). 

Then just find each function id, one after the other. Assert that they are both found.

Reply:
Fixed with `GetFunctionIterator` in `fuzzerutil`.